### DISCUSSION TITLE
Store no-go areas in favorites

### DISCUSSION THREAD

**Discussion by devemux86** [devemux86/cruiser#545] [Q&A] [answered]
Store [no-go areas](https://github.com/devemux86/cruiser/discussions/538) in favorites for easy management and use in route planning.

<img src="https://github.com/user-attachments/assets/b5ee2bef-53db-4866-8684-d9a74d9c66ad" width="50%"/><img src="https://github.com/user-attachments/assets/aa478a8a-1c0c-43d4-b508-2fa684170076" width="50%"/>

**Comment by DolomitesTom** [comment]
Sounds very good

**Comment by 0709wiwiwi** [comment]
As you start storing more and more favourites in the app itself, a decent backup facility becomes increasingly imperative.

**Reply by MotoUKRider** [reply]
Agreed.  Although the Profile Import/Export works 100% between devices - it's not immediate obvious *what* you're backing-up.  Something in the "Application" menu seems more logical.

**Reply by bios4** [reply]
I support the suggestion of a dedicated "backup & restore" (import/export, whatever) menu in the app instead of having multiple places to do so ("Profle", "My collection", ...)!

Is this something that can be implemented with a reasonable amount of effort @devemux86?

**Reply by devemux86** [reply]
@bios4 Probably, with a selection screen what to backup from the app:
favorites, routes, tracks, no-go areas, profiles, ... We have a topic here:
- https://github.com/devemux86/cruiser/discussions/556

**Comment by WolfgangZ42** [comment]
> Favorites are global for all profiles, so it not simple to change now their structure and separate them.

That is correct. But then the no-go areas have to be activated before each new route is created. 

I found the solution developed together with afischerdev about a year ago and the subsequent implementation in Cruiser 3.0.23 to be perfect (see https://github.com/devemux86/cruiser/discussions/85#discussioncomment-6161503). I told everyone who would listen about the uniqueness of Cruiser. To my knowledge, no other navigation software offers this option. **And I don't understand why this solution can't remain parallel to the "no-go-areas in route planning".** 

Imagine you define no-go-areas for the bycicle profile (e.g. stairways) you don't need in the walking profile. Or you'll avoid inner cities around your home address permanently, but only for motorcycle tours. 

As always: you decide and I can live with any solution ...

**Reply by devemux86** [reply]
> But then the no-go areas have to be activated before each new route is created.

This is not so simple, as no-go areas are stored with the route, like the waypoints.
If there is no route to save, we cannot save waypoints or no-go areas.

> why this solution can't remain parallel to the "no-go-areas in route planning".

@WolfgangZ42 Please see Cruiser [4.1.0-b2](https://github.com/devemux86/cruiser/discussions/544#discussioncomment-9530401).

> Imagine you define no-go-areas for the bycicle profile (e.g. stairways) you don't need in the walking profile. Or you'll avoid inner cities around your home address permanently, but only for motorcycle tours.

That's why we could probably use the favorite no-go areas, to easily add them in route planning.

**Reply by umpfmpf** [reply]
I think the option with the no-go areas is good, but I would welcome it if this were to be set as a global option for routes that should be avoided entirely plus individual areas only for the current route. 
And then the option not to drive a route twice?!?

**Comment by 0709wiwiwi** [comment]
No nogo areas, but single nogo streets.
Nogo track generated by Cruiser_Brouter internal.

*A nogo also works with a simple track file.

You can so block road sections per (motor)profile.
You can also just block known unpleasant roads.
Like a 30 km/h road in your neighbourhood to avoid.
You import the track(s) in the profile and that's it.
What is missing after import is a visual indication.
A nogo trackstyle xxxxxx (*block width) is desirable.
One import gpx file can contain several nogo tracks. 
[nogo_test_regio_donk.gpx.txt](https://github.com/devemux86/cruiser/files/15501366/nogo_test_regio_donk.gpx.txt)

Example. Red track overlay are blocked roads.
![image](https://github.com/devemux86/cruiser/assets/54167201/a7bbbdfa-4395-4d75-bdbe-9dd856dbdae7)

The result.
[Street_block_test_result.gpx.txt](https://github.com/devemux86/cruiser/files/15501617/Street_block_test_result.gpx.txt)

**Reply by devemux86** [reply]
Thanks @0709wiwiwi, a visual representation of the loaded external no-go files would be nice.

**Reply by bios4** [reply]
> What is missing after import is a visual indication.

You could achieve this (visually) by following the nice [approach for road closures](https://github.com/devemux86/cruiser/discussions/488#discussioncomment-9076793) by DIrgi.

But I agree that it would be great if such roads/tracks could be considered somehow automatically for routing and navigation.

There is the option in My collection > Favorites to define "Use in navigation" for a given favorite group.
Could such a behaviour - but in the exact opposite direction as "Do NOT use in navigation" - be implemented for tracks @devemux86?

**Reply by devemux86** [reply]
@bios4 thanks, this is an interesting idea, you mean "Do not use in routing".
However lines are not easy to exclude in routing, as they have too many points.

That's why the circle (center + radius) is the most simple form of no-go area.
Everything else is much more complicated to manage. :slightly_smiling_face:

**Reply by devemux86** [reply]
Alternatively, a no-go track can also be represented by a collection of small no-go circles with a radius of a few meters.

**Comment by devemux86** [comment]
No-go points / circles will be stored in favorites for easy management and use in route planning.

- I updated above the [1st post](https://github.com/devemux86/cruiser/discussions/545#discussion-6714350) with some new screenshots.

We will see what is possible with the complex no-go polygons.

**Comment by devemux86** [comment]
@0709wiwiwi 

We need to store the no-go circle radius in the GPX format.

Avoiding extensions, we can use the `hdop` field of the `wpt`.
(it is already defined as horizontal radius in meters)

**Comment by devemux86** [comment]
Store no-go circles in "Favorites" and no-go polygons in "Tracks / Areas" is available in:

- https://github.com/devemux86/cruiser/discussions/610

**Reply by bios4** [reply]
Is there a possibility to create a polygon no-go area within the app itself @devemux86?

Or are we limited to add only circle-shaped areas by long-pressing the map and have to add (externally created) polygons via import?

**Reply by devemux86** [reply]
> Is there a possibility to create a polygon no-go area within the app itself @devemux86?

Currently this is not possible, as polygon design on mobile is much more complicated.
So I have to see how it could be done in an optimal way.

[BRouter website](https://brouter.de/brouter-web/) can draw circular no-go areas and load external no-go polygons.
[OpenRouteService](https://maps.openrouteservice.org/) and [Valhalla](https://valhalla.openstreetmap.de/) websites can draw polygons.

> Or are we limited to add only circle-shaped areas by long-pressing the map and have to add (externally created) polygons via import?

Correctly, now we can:
- Draw circular no-go areas (point + radius)

- Import external polygons as no-go areas

**Reply by Ciutnik** [reply]
@devemux86 
> Currently this is not possible, as polygon design on mobile is much more complicated. So I have to see how it could be done in an optimal way.

We know the way to use the crosshair in the center of the screen. This is also the case in Locus when planning a path. In addition to the crosshair, two buttons are displayed: "+" for adding a point and "-" for deleting. In our case, the third one would be useful: "Finish and save". Or if we are only planning an area, adding a point close to the first one would mean closing the area and ending the drawing. 🙂

**Comment by devemux86** [comment, marked as answer]
- Store no-go areas in favorites in https://github.com/devemux86/cruiser/discussions/641

(points in "Favorites" and polygons in "Tracks / Areas")
